aiohttp/connector.py

```diff
@@ -770,6 +778,9 @@ def __init__(
         self._throttle_dns_events: Dict[Tuple[str, int], EventResultOrError] = {}
         self._family = family
         self._local_addr = local_addr
+        self._network_interface = network_interface
+        if self._network_interface and not SOCKET_SUPPORTS_BINDING_TO_DEVICE:
```

Comment:
@Dreamsorcerer what are your thoughts on having this as a hard assertion vs emitting a warning and falling back to a regular `bind()`? I think that `curl` does that implicitly. OTOH, it's a CLI app and not a framework (not sure if libcurl behaves the same, though).

Reply:
Hmm, I'm not sure. I think if I'm trying to bind to a specific interface, I probably want an error to tell me it's not going to happen. If I'd _prefer_ a specific interface, then a `try ... catch ...` to fall back would be fine, and it's explicit then too.

Reply:
Right, makes sense. @Mostafa-Mahdi would you mind incorporating this in the docs? A try-except example seems a natural for that FAQ doc you're already changing.

Reply:
Sure.

Reply:
Resolving this thread.